### PR TITLE
Create Form for Robust Accessorial Item Code 35A #163880118

### PR DESCRIPTION
**Pull request by donaldthai**
## Description

This PR adds a form for item code 35A so that we can capture more information in the pre-approval requests for Third Party Services description, reason for service and dollar amount. The feature flag `robustAccessorial` will be used for this item code as well.

`$` sign will now be prefixed to the dollar field. This affects all `cents` format.

## Code Review Verification Steps

* [ ] There are no aXe warnings for UI.
* [ ] This works in IE.
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/163880118) for this change

## Screenshots

image

image

### CODE REVIEW

File: src/shared/PreApprovalRequest/Code105Form.jsx [hunk 1]
```diff
@@ -6,7 +6,7 @@ export const Code105Form = props => {
   const { ship_line_item_schema } = props;
   return (
     <Fragment>
-      <SwaggerField fieldName="description" swagger={ship_line_item_schema} required />
+      <SwaggerField className="textarea-half" fieldName="description" swagger={ship_line_item_schema} required />
```
Comment (lynzt) [inline src/shared/PreApprovalRequest/Code105Form.jsx]: just curious what this is doing: `textarea-half`

File: src/shared/PreApprovalRequest/Code35Form.test.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  },
+};
+
+let wrapper;
+describe('code 35A details component', () => {
+  describe('renders', () => {
+    wrapper = shallow(<Code35Form ship_line_item_schema={simpleSchema} />);
+
+    it('without crashing', () => {
+      // eslint-disable-next-line
```
Comment (lynzt) [inline src/shared/PreApprovalRequest/Code35Form.test.js]: this commnet needed?

File: src/shared/PreApprovalRequest/DetailsHelper.js [hunk 3]
```diff
@@ -10,6 +11,8 @@ export function getFormComponent(code, robustAccessorial, initialValues) {
   const isNew = !initialValues;
   if (robustAccessorial && (code.startsWith('105b') || code.startsWith('105e'))) {
     if (isNew || hasCrateDimensions) return Code105Form;
+  } else if (robustAccessorial && code.startsWith('35')) {
```
Comment (lynzt) [inline src/shared/PreApprovalRequest/DetailsHelper.js]: We could do the `robustAccessorial` check at the top of the function: `if !robustAccessorial return DefautlForm` 
That way we only have to check the code in the `if / else if`...
Just a thought...

File: src/shared/PreApprovalRequest/Code105Form.jsx (hunk 1 above)
Comment (donaldthai) [reply src/shared/PreApprovalRequest/Code105Form.jsx]: That decreases the height of the input by half than it normally is.

File: src/shared/PreApprovalRequest/Code35Form.test.js (hunk 2 above)
Comment (donaldthai) [reply src/shared/PreApprovalRequest/Code35Form.test.js]: nope, will remove 👍

File: src/shared/PreApprovalRequest/DetailsHelper.js (hunk 3 above)
Comment (donaldthai) [reply src/shared/PreApprovalRequest/DetailsHelper.js]: ohh let me refactor

Comment (donaldthai) [reply src/shared/PreApprovalRequest/DetailsHelper.js]: @lynzt, going to push back on this since we need that flag for each item code for release.

**Pull request merged**